server/swap/swap.go

```diff
@@ -501,7 +501,7 @@ func (s *Swapper) restoreState(state *State, allowPartial bool) error {
 
 		if redeemCoin := ssd.RedeemCoinIn; len(redeemCoin) > 0 {
 			assetID := ssd.RedeemAsset
-			redeem, err := s.coins[assetID].Backend.Redemption(redeemCoin, swapCoin)
```

Comment:
These two coins were from different assets.  The swap coin was supposed to be the *counterparty's* swap, not the user's swap.  Test are updated to include the swap coin